docs/codeql/codeql-overview/codeql-changelog/codeql-cli-2.15.3.rst

```diff
@@ -0,0 +1,199 @@
+*   Fixed an issue where some Realm database sinks were not being recognized for the :code:`swift/cleartext-storage-database` query.
+
+New Queries
+~~~~~~~~~~~
+
+Swift
+"""""
+
+*   Added new query "System command built from user-controlled sources" (:code:`swift/command-line-injection`) for Swift. This query detects system commands built from user-controlled sources without sufficient validation. The query was previously `contributed to the 'experimental' directory by @maikypedia <https://github.com/github/codeql/pull/13726>`__ but will now run by default for all code scanning users.
+*   Added a nw query "Missing regular expression anchor" (:code:`swift/missing-regexp-anchor`) for Swift. This query detects regular expressions without anchors that can be vulnerable to bypassing.
+
+Query Metadata Changes
+~~~~~~~~~~~~~~~~~~~~~~
+
+JavaScript/TypeScript
+"""""""""""""""""""""
+
+*   Lower the severity of log-injection to medium.
+*   Increase the severity of XSS to high.
+
+Language Libraries
+------------------
+
+Bug Fixes
+~~~~~~~~~
+
+Golang
+""""""
+
+*   A bug has been fixed that meant that value flow through an array was not tracked correctly in some circumstances. Taint flow was tracked correctly.
+
+Breaking Changes
+~~~~~~~~~~~~~~~~
+
+C/C++
+"""""
+
+*   The expressions :code:`AssignPointerAddExpr` and :code:`AssignPointerSubExpr` are no longer subtypes of :code:`AssignBitwiseOperation`.
+
+Major Analysis Improvements
+~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Swift
+"""""
+
+*   Added Swift 5.9.1 support
+*   New AST node is extracted: :code:`SingleValueStmtExpr`
+
+Minor Analysis Improvements
+~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+C/C++
+"""""
+
+*   The "Returning stack-allocated memory" (:code:`cpp/return-stack-allocated-memory`) query now also detects returning stack-allocated memory allocated by calls to :code:`alloca`, :code:`strdupa`, and :code:`strndupa`.
+*   Added models for :code:`strlcpy` and :code:`strlcat`.
+*   Added models for the :code:`sprintf` variants from the :code:`StrSafe.h` header.
+*   Added SQL API models for :code:`ODBC`.
+*   Added taint models for :code:`realloc` and related functions.
+
+C#
+""
+
+*   The predicate :code:`UnboundGeneric::getName` now prints the number of type parameters as a :code:` `N` suffix, instead of a :code:`<,...,>` suffix. For example, the unbound generic type
+    :code:`System.Collections.Generic.IList<T>` is printed as :code:`IList`1` instead of :code:`IList<>`.
+*   The predicates :code:`hasQualifiedName`, :code:`getQualifiedName`, and :code:`getQualifiedNameWithTypes` have been deprecated, and are instead replaced by :code:`hasFullyQualifiedName`, :code:`getFullyQualifiedName`, and :code:`getFullyQualifiedNameWithTypes`, respectively. The new predicates use the same format for unbound generic types as mentioned above.
+*   These changes also affect models-as-data rows that refer to a field or a property belonging to a generic type. For example, instead of writing
```

Comment:
I don't believe that we've introduced users to the "models-as-data" terminology yet. We also haven't shipped CodeQL model packs for languages other than Java yet. Not sure that this is a user-facing detail 🤔 

Also, I thought that our main intention was to get people generating model packs using the VS Code extension not writing things by hand.

Reply:
This is a slightly grey area because the 'models as data' in the C# are clearly user-facing but we won't officially support CodeQL model packs for C# until the model editor works for C#. However this is a minor analysis change that only the most curious of users will engage with so I think this changenote is ok. 

cc @coadaflorin just in case you have a different view.

Reply:
My point was mostly that if we haven't published information to users about these rows, then generally they shouldn't have a change note.

Reply:
But happy to leave it there if you think it's useful to users.

Reply:
If writing MaD specifications for C# is not even (un)officially supported, we can drop the change note.

Reply:
> However this is a minor analysis change that only the most curious of users will engage with so I think this changenote is ok.

FTR, I also think it's OK to have here. If we get questions about it, we can re-evaluate.

Reply:
Thanks. Something else seems to be off: There is missing something after `For example, instead of writing` in https://codeql.github.com/docs/codeql-overview/codeql-changelog/codeql-cli-2.15.3/#id7.

Reply:
Can you point us at the original source file, so we can work out what's missing and what went wrong with the generator?

Reply:
It's https://raw.githubusercontent.com/github/codeql/main/csharp/ql/lib/change-notes/released/0.8.3.md

I found the bug and will fix it next week (in the meantime, another temp fix in the generated files is OK). Turns out this changenote is a great test case Tom 😆

Reply:
Fixed as part of https://github.com/github/codeql/pull/14889 - I discovered a handful of other bugs based on this investigation